### PR TITLE
feat: refactor RocketMQ bridge to connector and action

### PR DESCRIPTION
**Pull request by kjellwinblad**
Fixes:
https://emqx.atlassian.net/browse/EMQX-11467

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [] Added property-based tests for code which performs user input validation
- [] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl [hunk 1]
```diff
@@ -58,15 +61,144 @@ values(post) ->
             max_buffer_bytes => ?DEFAULT_BUFFER_BYTES
         }
     };
-values(put) ->
-    values(post).
+conn_bridge_example_values(put) ->
+    conn_bridge_example_values(post).
+
+%% TODO fix these examples
```
Comment (thalesmg) [inline apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl:67]: Is this TODO still applicable?

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl [hunk 2]
```diff
@@ -94,6 +226,16 @@ desc("config") ->
     ?DESC("desc_config");
 desc(Method) when Method =:= "get"; Method =:= "put"; Method =:= "post" ->
     ["Configuration for RocketMQ using `", string:to_upper(Method), "` method."];
+desc("creation_opts") ->
+    ?DESC(emqx_resource_schema, "creation_opts");
```
Comment (thalesmg) [inline apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl:230]: Nit: I couldn't spot a `fields("creation_opts")` clause :see_no_evil:

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl [hunk 3]
```diff
@@ -94,18 +103,18 @@ on_start(
...
-        sync_timeout => SyncTimeout,
-        templates => Templates,
-        producers_opts => ProducerOpts
+        acl_info => ACLInfo,
+        installed_channels => #{}
+        % topic => Topic,
+        % topic_tokens => TopicTks,
+        % sync_timeout => SyncTimeout,
+        % templates => Templates,
+        % producers_opts => ProducerOpts
```
Comment (thalesmg) [inline apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl:117]: :scissors: ?

**Review by thalesmg — APPROVED** [review]
Nit: might be good to have some smoke tests that go through the v2 APIs and schemas (like the ones in `emqx_bridge_v2_testlib`).

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl (hunk 2 above)
Comment (kjellwinblad) [reply apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl]: Fixed: [976099f](https://github.com/emqx/emqx/pull/12488/commits/976099f5fbb3f1894b77d9017465bef114ff4eef)

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl (hunk 1 above)
Comment (kjellwinblad) [reply apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq.erl]: Fixed: [976099f](https://github.com/emqx/emqx/pull/12488/commits/976099f5fbb3f1894b77d9017465bef114ff4eef)

**Comment by kjellwinblad** [comment]
Thanks @thalesmg for a good review. I fixed the issues that you pointed out.

**Comment by kjellwinblad** [comment]
> Nit: might be good to have some smoke tests that go through the v2 APIs and schemas (like the ones in emqx_bridge_v2_testlib).

Thanks. I will have a look at emqx_bridge_v2_testlib. If we add such tests I think we can do it in another PR so we can get this merged.

**Review by ieQu1 — APPROVED** [review]

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl [hunk 4]
```diff
@@ -123,6 +127,64 @@ on_start(
...
+    %% Update state
+    NewState = OldState#{installed_channels => NewInstalledChannels},
+    {ok, NewState}.
+
+on_get_channel_status(
+    _ResId,
+    _ChannelId,
+    _State
+) ->
+    ?status_connected.
```
Comment (zhongwencool) [inline apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl:183]: check channel ID exist.

**Pull request merged**